databricks/koalas/series.py

```diff
@@ -3227,6 +3227,86 @@ def copy(self) -> 'Series':
         """
         return _col(DataFrame(self._internal.copy()))
 
+    def mode(self, dropna=True) -> 'Series':
+        """
+        Return the mode(s) of the dataset.
+
+        Always returns Series even if only one value is returned.
+
+        Parameters
+        ----------
+        dropna : bool, default True
+            Don't consider counts of NaN/NaT.
+
+        Returns
+        -------
+        Series
+            Modes of the Series in sorted order.
+
+        Examples
+        --------
+        >>> s = ks.Series([0, 0, 1, 1, 1, np.nan, np.nan, np.nan])
+        >>> s
+        0    0.0
+        1    0.0
+        2    1.0
+        3    1.0
+        4    1.0
+        5    NaN
+        6    NaN
+        7    NaN
+        Name: 0, dtype: float64
+
+        >>> s.mode()
+        0    1.0
+        Name: 0, dtype: float64
+
+        If there are several same modes, all items are shown
+
+        >>> s = ks.Series([0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3,
+        ...                np.nan, np.nan, np.nan])
+        >>> s
+        0     0.0
+        1     0.0
+        2     1.0
+        3     1.0
+        4     1.0
+        5     2.0
+        6     2.0
+        7     2.0
+        8     3.0
+        9     3.0
+        10    3.0
+        11    NaN
+        12    NaN
+        13    NaN
+        Name: 0, dtype: float64
+
+        >>> s.mode()
+        0    1.0
+        1    2.0
+        2    3.0
+        Name: 0, dtype: float64
+
+        With 'dropna' set to 'False', we can also see NaN in the result
+
+        >>> s.mode(False)
+        0    NaN
+        1    1.0
+        2    2.0
+        3    3.0
+        Name: 0, dtype: float64
+        """
+        ser_count = self.value_counts(dropna=dropna)
+        sdf_count = ser_count._internal.sdf
+        most_value = sdf_count.head(1)[0][1]
+        sdf_most_value = (sdf_count.where("count == {}".format(most_value))
+                                   .sort('__index_level_0__'))
```

Comment:
Do we need `sort` here?

Reply:
i think not necessary, thanks! :)

Reply:
ah, we might need to sort the result values since the doc says it returns values in sorted order. [here](https://github.com/databricks/koalas/pull/899/files#diff-f38e425003958ac0d1bd6ffa418f2613R3240), also https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.mode.html Otherwise we should remove "in sorted order" from the doc.
See #1047
cc @HyukjinKwon

Reply:
In my opinion, maybe it would be better sort the order like pandas for now, since they wrote down explicitly "sorted order" only for `Series.mode()`.
btw, For those people who prefer to see sorted results, maybe we can make an option value to make possible to choose whether we sort the result or not?

Reply:
let's fix it to sort first and see if people actually need it and report an issue.
Thanks @ueshin. I missed https://github.com/databricks/koalas/pull/899#discussion_r346011895